### PR TITLE
Add retry to waitFor

### PR DESCRIPTION
**Pull request by pkoch**
Nothing fancy.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -71,7 +83,10 @@ const main = async () => {
     core.setOutput("task-arn", taskArn);
 
     core.debug("Waiting for task to finish...");
-    await ecs.waitFor("tasksStopped", { cluster, tasks: [taskArn] }).promise();
+    await retry_promise(
+      3,
+      () => ecs.waitFor("tasksStopped", { cluster, tasks: [taskArn] }).promise(),
+    )
```
Comment (Strnadj) [inline index.js]: did you check the documentation? There is something called as https://github.com/aws/aws-sdk-js/blob/307e82673b48577fce4389e4ce03f95064e8fe0d/lib/service.d.ts#L53
there is maybe a high chance, that you could pass `$waiter` there and just extend those limits... I will be expecting that this function will have some similar arguments.

**Review by Strnadj — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (Strnadj) [reply index.js]: https://github.com/aws/aws-sdk-js/blob/cf8fe5adad400bb6a918843aed0f80ee47fe3a48/clients/ecs.d.ts#L434 this exactly

Comment (pkoch) [reply index.js]: I had checked the docs (not the source), and I didn't see this there. Great catch. 👌

Looking at https://github.com/aws/aws-sdk-js/blob/cf8fe5adad400bb6a918843aed0f80ee47fe3a48/lib/service.d.ts#L6, I agree we should just raise `maxAttempts`. Digging for the current value.

Comment (pkoch) [reply index.js]: Current [is 100](https://github.com/aws/aws-sdk-js/blob/2e5ecc6fe766f0aff1c8dd482871e626cf451869/apis/ecs-2014-11-13.waiters2.json#L32). Making the PR go for 300.

File: index.js [hunk 2]
```diff
@@ -71,7 +71,11 @@ const main = async () => {
     core.setOutput("task-arn", taskArn);
 
     core.debug("Waiting for task to finish...");
-    await ecs.waitFor("tasksStopped", { cluster, tasks: [taskArn] }).promise();
+    await ecs.waitFor("tasksStopped", {
+      cluster,
+      tasks: [taskArn],
+      $waiter: { maxAttempts: 300 },
+    }).promise();
```
Comment (pkoch) [inline index.js:78]: From what I can see in their `.d` file, this has the signature:
```
waitFor(state: "tasksStopped", params: ECS.Types.DescribeTasksRequest & {$waiter?: WaiterConfiguration}, callback?: (err: AWSError, data: ECS.Types.DescribeTasksResponse) => void): Request<ECS.Types.DescribeTasksResponse, AWSError>;
```

So, I presume this should work. How do you think we should test this change?

Comment (Strnadj) [reply index.js:78]: I am curious what is the default value, which is currently insufficient.

**Review by Strnadj — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (pkoch) [reply index.js:78]: > https://github.com/trustfractal/aws-ecs-run-task/pull/1#discussion_r1015574967

100 attempts, waiting 6 (seconds, I presume) between attempts. So, 10 minutes, which is what we've been observing.

Comment (pkoch) [reply index.js:78]: How do you think we should test this?

**Review by Strnadj — APPROVED** [review]

**Pull request merged**